sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroSource.java

```diff
@@ -393,7 +393,7 @@ private static InputStream decodeAsInputStream(byte[] data, String codec) throws
       ByteArrayInputStream byteStream = new ByteArrayInputStream(data);
       switch (codec) {
         case DataFileConstants.SNAPPY_CODEC:
-          return new SnappyCompressorInputStream(byteStream);
+          return new SnappyCompressorInputStream(byteStream, 1 << 16 /* Avro uses 64KB blocks */);
```

Comment:
It seems easier to adapt to the interface they expose then to re-implement.

It seems as though internally in Avro they just access the byte buffer array directly. Should be easy enough to wrap byte[] into a ByteBuffer and unwrap to create a byte[] and or use a ByteArrayInputStream like equivalent.

Also, it seems as though the Codec for Snappy includes a CRC32 checksum which we are ignoring that we could benefit from if we used the Codec directly (https://avro.apache.org/docs/1.7.7/spec.html#snappy)